services/agent/src/simcore_service_agent/modules/task_monitor.py

```diff
@@ -101,57 +106,100 @@ def register_job(
 
-        self._to_start[target.__name__] = task_data
+    async def unregister_job(self, target: Callable) -> None:
+        job_name = target.__name__
+        task: Optional[asyncio.Task] = self._tasks.get(job_name, None)
+        if task is not None:
+            await self._cancel_task(task)
+            del self._tasks[job_name]
+        self._to_start.pop(job_name, None)
 
     async def start(self) -> None:
-        self._was_started = True
-        for name, task_data in self._to_start.items():
+        """schedule tasks for all jobs"""
+        for name in self._to_start:
             logger.info("Starting task '%s'", name)
-            self._tasks.add(
-                asyncio.create_task(_task_runner(task_data), name=f"task_{name}")
-            )
+            self.start_job(name)
+        self._was_started = True
 
-    async def shutdown(self):
+    @staticmethod
+    async def _cancel_task(task: asyncio.Task) -> None:
         async def _wait_for_task(task: asyncio.Task) -> None:
             with suppress(asyncio.CancelledError):
                 await task
 
-        tasks_to_wait: deque[Awaitable] = deque()
-        for task in set(self._tasks):
-            logger.info("Cancel and stop task '%s'", task.get_name())
+        logger.info("Cancel task '%s'", task.get_name())
+        task.cancel()
+        await _wait_for_task(task)
 
-            task.cancel()
-            tasks_to_wait.append(_wait_for_task(task))
-            self._tasks.remove(task)
+    async def shutdown(self):
+        await asyncio.gather(
+            *(self._cancel_task(t) for t in self._tasks.values()),
+            return_exceptions=True,
+        )
 
-        await asyncio.gather(*tasks_to_wait, return_exceptions=True)
         self._was_started = False
-        self._to_start = {}
+        self._tasks.clear()
+        self._to_start.clear()
 
 
-def setup(app: FastAPI) -> None:
-    async def _on_startup() -> None:
-        task_monitor = app.state.task_monitor = TaskMonitor()
-        settings: ApplicationSettings = app.state.settings
+async def disable_volume_removal_task(app: FastAPI) -> None:
+    task_monitor: TaskMonitor = app.state.task_monitor
 
-        # setup all relative jobs
+    await task_monitor.unregister_job(backup_and_remove_volumes)
+    logger.debug(
+        "Disabled '%s' job.",
+        backup_and_remove_volumes.__name__,
+    )
+
+
+async def enable_volume_removal_task_if_missing(app: FastAPI) -> None:
+    task_monitor: TaskMonitor = app.state.task_monitor
+    settings: ApplicationSettings = app.state.settings
+
+    try:
         task_monitor.register_job(
             backup_and_remove_volumes,
-            settings,
+            app,
             repeat_interval_s=settings.AGENT_VOLUMES_CLEANUP_INTERVAL_S,
         )
+        if task_monitor.start_job(backup_and_remove_volumes.__name__):
+            logger.debug(
+                "Enabled '%s' job.",
+                backup_and_remove_volumes.__name__,
```

Comment:
so you have a background task that already removes volumeS?
would it not be easier to trigger the task intead of creating all this stuff with the low priority manager etc?? I find this very complex.